STYLE_GUIDE.md

```diff
@@ -0,0 +1,1245 @@
+}
+```
+
+</details>
+
+When writing a spec with multiple files, make the files other than the import
+partials unless there's a concrete reason not to. This helps make it clearer at
+a glance that they're additional input files, rather than expectation files.
+
+### DO name single imported or used files "other"
+
+<details>
+<summary>Example</summary>
+
+#### Good
+
+```hrx
+<===> member/variable/input.scss
+@import "other";
+
+a {b: $c}
+
+<===> member/variable/_other.scss
+$c: d;
+
+<===> member/variable/output.scss
+a {
+  b: d;
+}
+```
+
+#### Bad
+
+```hrx
+<===> member/variable/input.scss
+@import "imported";
+
+a {b: $c}
+
+<===> member/variable/_imported.scss
+$c: d;
+
+<===> member/variable/output.scss
+a {
+  b: d;
+}
+```
+
+</details>
+
+When a spec involves exactly two files, the file that's not `input.scss` should
+be named "other" (`_other.scss`, `_other.sass`, etc).
+
+### PREFER referring to issues when marking specs as `:todo`
+
+<details>
+<summary>Example</summary>
+
+#### Good
+
+```hrx
+<===> slash_slash_string/options.yml
+---
+:todo:
+- libsass # sass/libsass#2840
+
+<===> slash_slash_string/input.scss
+a {b: 1 / 2 / foo()}
+
+<===> slash_slash_string/output.css
+a {
+  b: 1/2/foo();
+}
+```
+
+#### Bad
+
+```hrx
+<===> slash_slash_string/options.yml
+---
+:todo:
+- libsass
+
+<===> slash_slash_string/input.scss
+a {b: 1 / 2 / foo()}
+
+<===> slash_slash_string/output.css
+a {
+  b: 1/2/foo();
+}
+```
+
+</details>
+
+When [marking a spec as `:todo`][], follow it with a comment that refers to an
+issue in the implementation in question tracking the feature the spec tests. If
+there isn't an issue yet, create one. Make sure to also link that issue back to
+the specs for it!
+
+[marking a spec as "todo"]: README.md#todo
+
+## Documentation
+### CONSIDER adding a comment explaining your spec
+
+<details>
+<summary>Example</summary>
+
+#### Good
+
+```
+<===> empty/input.scss
+// CSS requires at least one token in a custom property.
+.empty {
+  --property:;
+}
+
+<===> empty/error
+Error: Expected token.
+  ,
+3 |   --property:;
+  |              ^
+  '
+  /sass/spec/css/custom_properties/error/empty/input.scss 3:14  root stylesheet
+```
+
+</details>
+
+If the details of what your spec is testing and why it produces the output it
+does wouldn't be clear to someone moderately familiar with Sass from [the spec's
+path][] and a casual reading of the spec's contents, it's a good idea to add a
+comment explaining in detail what's going on. This can also be useful for
+elucidating the stranger corners of the CSS spec or Sass's language semantics.
+
+[the spec's path]: #do-use-descriptive-paths
+
+Explanatory comments should always be silent Sass comments. Loud comments and
+HRX comments must not be used.
+
+### DON'T write HRX comments
+
+<details>
+<summary>Example</summary>
+
+#### Bad
+
+```
+<===>
+CSS requires at least one token in a custom property.
+<===> empty/input.scss
+.empty {
+  --property:;
+}
+
+<===> empty/error
```

Comment:
This should be error/empty, because this bad example is for comments (and to be consistent with the path on line 788).

Reply:
This is notionally within an `error.hrx` file (see the stack trace below), so I think it's fine as-is. Note that the spec that explicitly includes `error/` is doing so to demonstrate it as a sibling of a non-error spec.